lib/pups/config.rb

```diff
@@ -4,49 +4,86 @@ module Pups
   class Config
     attr_reader :config, :params
 
-    def initialize(config, ignored = nil)
+    def initialize(
+      config,
+      ignored = nil,
+      tags: tags = nil,
+      skip_tags: skip_tags = nil
+    )
       @config = config
 
       # remove any ignored config elements prior to any more processing
       ignored&.each { |e| @config.delete(e) }
 
+      filter_tags(tags)
+      filter_tags(skip_tags, true)
```

Comment:
sig on this method is a bit odd: 

Maybe ... 

filter_tags(include_tags: ... , exclude_tags: ... )

Reply:
Updated, it's a bit more verbose now but hopefully a little more readable